staging/src/k8s.io/client-go/tools/record/events_cache.go

```diff
@@ -362,11 +471,11 @@ func NewEventCorrelator(clock clock.Clock) *EventCorrelator {
 
 // EventCorrelate filters, aggregates, counts, and de-duplicates all incoming events
 func (c *EventCorrelator) EventCorrelate(newEvent *v1.Event) (*EventCorrelateResult, error) {
+	aggregateEvent, ckey := c.aggregator.EventAggregate(newEvent)
```

Comment:
How expensive is EventAggregate?  I assume that's why it was lower before - maybe we should filter twice, instead of once?